### PR TITLE
Add target option and test suite

### PR DESCRIPTION
**Pull request by matks**
This PR does multiple things:
1. It adds a `--target` option to Header Stamp to allow users to run it against a specific folder (currently it targets the current directory) ; default remains current directory
2. I did not want to add a new option without a reliable system to make sure I did not break anything. So I added tests
   - I added a small integration testing environment*
   - I used gsitemap module sample in this environment
   - I added the test in `.travis.yml`
3. I wrote a small README in `tests/integration` to explain how this test suite works.

This test suite is custom. No Behat, no PhpUnit. Because I think none of these frameworks fits current need. So I reused what I did for https://github.com/matks/ps-modules-builder which is a plain php script that does folder-to-folder comparaisons. The idea is simple: pick a module, run the tool against it, compare to "expected", fail if there are differences.

### CODE REVIEW

**Comment by matks** [comment]
Aaaaaand the test ran successfully in Travis 😄 https://travis-ci.com/github/PrestaShopCorp/header-stamp/jobs/355086451

```
$ php tests/integration/runner/run.php
 - module gsitemap processed successfully
Integration tests run successfully
```

**Comment by matks** [comment]
😭 My "test" PR succeeded although it should have failed, which indicates an issue in my test.

I get
```
Test failed for module gsitemap, got differences between expected folder and workspace folder :
 - /gsitemap-cron.php has different md5
The command "php tests/integration/runner/run.php" exited with 0.
```

What I want:
```
Test failed for module gsitemap, got differences between expected folder and workspace folder :
 - /gsitemap-cron.php has different md5
The command "php tests/integration/runner/run.php" exited with 1.
```

**Comment by matks** [comment]
Great 😁 learnt something today !

https://stackoverflow.com/a/40245330/8071080

`return` doesn't send any output to STDOUT, so I needed to use `exit` instead

File: tests/integration/runner/run.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+$filesystem = new Filesystem();
+$folderComparator = new FolderComparator();
+
+$application = buildTestApplication();
+
+foreach ($modulesToTest as $moduleName => $config) {
+    ++$workspaceID;
+    $moduleFolderpath = __DIR__ . '/../module-samples/' . $moduleName;
+    $expectedModuleFolderpath = __DIR__ . '/../expected/' . $moduleName;
+    $workspaceFolderpath = __DIR__ . '/../workspace/' . $workspaceID;
```
Comment (Quetzacoalt91) [inline tests/integration/runner/run.php:26]: Do you think we should ignore this folder?

Comment (matks) [reply tests/integration/runner/run.php:26]: Yes we should. I put it in my .gitignore at some point but it seems I got it wrong and it was not pushed

.gitignore should contain tests/integration/workspace/* (but the .gitkeep inside it should be into git)

Comment (matks) [reply tests/integration/runner/run.php:26]: Fixed in https://github.com/PrestaShopCorp/header-stamp/pull/19/commits/9e5a9d5fbcd7a60fcf6cddcc1806f85dd7448d79

**Comment by matks** [comment]
🆙 @Quetzacoalt91 @Matt75 🙏 ?

**Comment by Quetzacoalt91** [comment]
Thank you @matks

**Pull request merged**